### DISCUSSION TITLE
Question: Does Positron limit the number of CPU cores?

### DISCUSSION THREAD

**Discussion by fnincao** [posit-dev/positron#5095] [Q&A]
Hi all,

Thanks for this really amazing IDE you're building! I was trying to run some pipelines that need rstudioapi on my lab's server and was only able to do it with Positron (after many failed attempts). However, I’ve been limited to using only 2 CPU cores from the server. I messaged the IT team, and they assured me there was no limitation from the hardware side, suggesting that the software might be the cause. Therefore, does Positron limit the number of cores used in parallelized code?

Thanks!

Sincerely,
Felipe

**Comment by sharon-wang** [comment]
Hi @fnincao, thank you for reaching out!

There shouldn't be any limitations from Positron on the number of cores, but let's try to see where this limitation is coming from. Would you mind sharing a bit more about your environment, such as the operating system of the server and the parallelization library you're using for the pipelines? 

Are you seeing the same behaviour (limitation to 2 cores) when running the pipelines on the same server, but outside of Positron, e.g. when running the code directly in a vanilla R or Python session?

If you're running on a Linux machine, the output of `ulimit -a` from inside an R or Python session would also be helpful.

**Comment by fnincao** [comment]
Hi @sharon-wang, thanks for your reply!

I'm working on a Linux server, and the library I'm using is LAStools (https://github.com/LAStools/LAStools). I am also running the same pipeline in a Windows environment with the same library, where I can use all available cores. Below are the error messages I'm getting in the console:

<img width="1258" alt="Screenshot 2024-10-23 at 14:19:02" src="https://github.com/user-attachments/assets/d1d0b737-ccb1-4d34-a2b3-8ebab105e435"> <img width="405" alt="Screenshot 2024-10-23 at 14:20:29" src="https://github.com/user-attachments/assets/27e0950b-0021-48a4-827a-dc246f1a4779">
Additionally, here is the output of the ulimit -a command from my console:

<img width="417" alt="Screenshot 2024-10-23 at 14:29:33" src="https://github.com/user-attachments/assets/3b44bed8-6480-41c0-84f5-f4a3d6dcfe43">

Thanks again for your help.

Sincerely,
Felipe

**Reply by sharon-wang** [reply]
Thanks for the additional info!

From the `ulimit` output, nothing really sticks out, so we can rule out system resource limits on your user.

Some more questions for you:

- If you run the pipeline outside of Positron, are you seeing the same warning about core limitations on your Linux server? On your Windows machine, are you running within Positron?

- Is it possible that you're running the pipeline a bit differently on your Linux server versus your Windows environment? Perhaps the setting mentioned in the output `n_cores` is set differently in the Linux environment?

**Reply by sharon-wang** [reply]
Hey @fnincao, just wanted to check in and see if you were able to get past this issue?
